mathesar/api/viewsets/db_types.py

```diff
@@ -0,0 +1,18 @@
+from rest_framework import viewsets
+from rest_framework.response import Response
+from db.types.base import db_types_hinted, PostgresType, MathesarCustomType
+from itertools import chain
+
+
+class DbTypeViewSet(viewsets.ViewSet):
+    def list(self, _):
+        known_vanilla_types = (postgres_type for postgres_type in PostgresType)
+        known_custom_types = (mathesar_custom_type for mathesar_custom_type in MathesarCustomType)
+        known_db_types = chain(known_vanilla_types, known_custom_types)
+        data = [
+            {
+                "id": db_type.value,
+                "hints": db_types_hinted.get(db_type, None)
```

Comment:
What is this hint expected to be used for?

Reply:
When a `DBFunction` has a hint with id `boolean` or `string_like`, the API client can check against the `db_types` endpoint to tell which database types are meant by that (which db types have the hints `boolean` or `string_like`).